src/components/postTags/PostTagForm.js

```diff
@@ -10,37 +10,37 @@ export const PostTagForm = ({postId, endEditTags}) => {
 	const { tags, getTags } = useContext(TagContext)
 	const { getPostTagsByPostId, addPostTag, deletePostTag } = useContext(PostTagContext)
 	const [ thisPostTags, setThisPostTags ] = useState([]) 
-	const [ selectedPostTags, setSelectedPostTags ] = useState([]) 
+  const [ selectedPostTags, setSelectedPostTags ] = useState([]) 
+  const [ isSubmitting, setIsSubmitting ] = useState(false)
 
 	const handleChange = (val) => {
 		setSelectedPostTags(val)
 	}
 
 	const savePostTags = () => {
-		
-		// Iterate through the selected tag ids and save any tags that weren't initially selected.
-		selectedPostTags.forEach( selectedTagId => {
-			const matchingPostTag = thisPostTags.find( pt => pt.tag_id === selectedTagId)
-			if (matchingPostTag === undefined) {
-				const newPostTag = {
-					post_id : postId,
-					tag_id : selectedTagId
-				}
-				addPostTag(newPostTag)
-			}
-		})
+    setIsSubmitting(true)
 
-		// Iterate through the post tags that were initially selected, and delete the ones that were removed.
-		thisPostTags.forEach( pt => {
-			const matchingSelection = selectedPostTags.find( selectedTagId => selectedTagId === pt.tag_id )
-			if (matchingSelection === undefined) {
-				deletePostTag(pt.id)
-			}
+    // Iterate through the selected tag ids and save any tags that weren't initially selected.
+    const addPostTagPromises = selectedPostTags
+      .filter(selectedTagId => !thisPostTags.some(pt => pt.tag_id === selectedTagId))
+      .map(selectedTagId => {
+        const newPostTag = {
+          post_id: postId,
+          tag_id: selectedTagId
+        };
 
-	})
-	endEditTags()
+        return addPostTag(newPostTag)
+      })
 
+    // Iterate through the post tags that were initially selected, and delete the ones that were removed.
+    const deletePostTagPromises = thisPostTags
+      .filter(pt => !selectedPostTags.some(selectedTagId => selectedTagId === pt.tag_id))
+      .map(pt => deletePostTag(pt.id))
 
```

Comment:
Similarly to adding tags, I would find this more readable with more named variables.

```
const postTagsToDelete = thisPostTags
      .filter(pt => !selectedPostTags.some(selectedTagId => selectedTagId === pt.tag_id))
const deletePostTagPromises = postTagsToDelete.map(pt => deletePostTag(pt.id))
```